src/Http/Http.Results/src/ResultsOfTHelper.cs

```diff
@@ -1,21 +1,51 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
+using System.Runtime.CompilerServices;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http.Metadata;
 
 namespace Microsoft.AspNetCore.Http;
 
 internal static class ResultsOfTHelper
 {
+    public const DynamicallyAccessedMemberTypes RequireMethods = DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods;
```

Comment:
Why are we marking these helpers as safe?

Reply:
To make all the ResultsOfT safe.

Reply:
What does that mean? Who is calling into it from safe code?

Reply:
Infer metadata calls `PopulateMetadata` on the result in this situation:
```csharp
app.MapGet("/todo/{id}", Results<Ok<Todo>, NotFound, BadRequest> (int id) => id switch
    {
        <= 0 => TypedResults.BadRequest(),
        >= 1 and <= 10 => TypedResults.Ok(new Todo(id, "Walk the dog")),
        _ => TypedResults.NotFound()
    });
```

Reply:
It won't call this from the source generator.